camus-etl-kafka/src/test/java/com/linkedin/camus/etl/kafka/CamusJobTestWithMock.java

```diff
@@ -170,17 +181,21 @@ private void setupJob1() {
     TopicMetadataResponse metadataResponse = EasyMock.createMock(TopicMetadataResponse.class);
     _mocks.add(metadataResponse);
     EasyMock.expect(metadataResponse.topicsMetadata()).andReturn(topicMetadatas).anyTimes();
+    return metadataResponse;
+  }
 
-    // For OffsetResponse
+  private OffsetResponse mockOffsetResponse(List<MyMessage> myMessages) {
     OffsetResponse offsetResponse = EasyMock.createMock(OffsetResponse.class);
     _mocks.add(offsetResponse);
-    List<MyMessage> myMessages = messagesWritten.get(TOPIC_1);
     // The first call is getLatestOffset, we set the value to 1
     EasyMock.expect(offsetResponse.offsets(EasyMock.anyString(), EasyMock.anyInt())).andReturn(new long[]{myMessages.size()}).times(1);
     // The second call is getEarliestOffset, we set the value to 0
     EasyMock.expect(offsetResponse.offsets(EasyMock.anyString(), EasyMock.anyInt())).andReturn(new long[]{0}).times(1);
+    EasyMock.expect(offsetResponse.hasError()).andReturn(false).anyTimes();
```

Comment:
Some tests may have more complex requirements, e.g., first two times has error, third time succeeds. Also, some tests may need to change the return value of a different method, rather than hasError(). 
So, I think there should be a separate method for each need. Currently there are 3: mockOffsetResponse(), mockOffsetResponseWithError() and mockOffsetResponseThridRetrySucceed().